docs/Streams/index.md

```diff
@@ -0,0 +1,73 @@
+# Streams
+
+A stream is a doubly linked list of bytes. It can be used for file input/output, DayZ serializers, hex conversion or string parsing. Once you are finished with the stream it is recommended to close the stream. This is done by using the `Close` method but the process automatically happens in the destructor.
+
+Streams can be manipulated with `CF_TextWriter`, and `CF_BinaryWriter`. Streams can be read with `CF_TextReader`, and`CF_BinaryReader`.
+
+In binary format, all values are in little-endian format.
+
+## Files
+
+Use the `CF_FileStream` class to read and write files on the a file system. 
+
+```csharp
+string path = "$profile:test.txt";
+
+// Create the file
+CF_TextWriter writer = new CF_TextWriter(new CF_FileStream(path, FileMode.WRITE));
+writer.WriteLine("This is some text");
+writer.Close();
+
+// Open the file and read it back
+CF_TextReader reader = new CF_TextReader(new CF_FileStream(path, FileMode.READ));
+Print(reader.ReadLine()); // "This is some text"
+reader.Close();
+```
+
+## String
+
+Use the `CF_StringStream` class to read and write to a string source. This stream does not accept '\0' bytes.
+
+```csharp
+string data = "Hello, World!";
+
+CF_TextReader reader = new CF_TextReader(new CF_StringStream(data));
+Print("ReadWord: " + reader.ReadWord());				// "Hello"
+Print("ReadChar: " + reader.ReadChar());				// ","
+Print("ReadWhitespace: " + reader.ReadWhitespace());	// ""
+Print("ReadWord: " + reader.ReadWord());				// "World"
+Print("ReadChar: " + reader.ReadChar());				// "!"
+reader.Close();
+```
```

Comment:
Maybe you could give a bit of information on how we detect what a word is? I would assume it reads until the next separator char? If so a quick list of which chars for this would be nice. Or if it's the rule "characters that are not aA-zZ" then say that.